pages/forums/view.js

```diff
@@ -23,41 +27,76 @@ const pageTitle = 'Public Forums'
 
 
 class ViewThread extends Component {
+  /***************************************************************************\
+    Private  Methods
+  \***************************************************************************/
+
+  _handleNewComment (comment) {
+    this.setState({
+      comments: [...this.state.comments, comment.data],
+    })
+  }
+
+  async _getComments (page, oldPage) {
+    let newState = { ...this.state }
+    const { getThreadComments, query } = this.props
+
+    if (page !== oldPage) {
+      const { payload, status } = await getThreadComments(query.id, page)
+
+      if (status === 'success') {
+        newState = {
+          comments: payload.data || [],
+          totalCommentPages: Math.ceil(payload.meta.total / payload.meta.limit),
+        }
+      } else {
+        newState.comments = []
+      }
+    }
+
+    return newState
+  }
+
   /***************************************************************************\
     Public Methods
   \***************************************************************************/
 
   async componentDidMount () {
```

Comment:
I thought we had moved this into `getInitialProps`? 🤔

Reply:
I can't actually _move_ it without the page breaking due to the double mounting issue. Same holds true for the group page from what I've seen in testing.

Reply:
Fixed the root cause of the issue. We should probably keep it there JUST in case, though.